testing/test_pytester.py

```diff
@@ -453,3 +454,15 @@ def test_timeout():
     )
     with pytest.raises(testdir.TimeoutExpired):
         testdir.runpytest_subprocess(testfile, timeout=1)
+
+
+def test_linematcher_with_set():
+    """Test LineMatcher with regard to passing in a set (accidentally)."""
+    lm = LineMatcher([])
+
+    with pytest.raises(AssertionError):
+        lm.fnmatch_lines({})
```

Comment:
this is a dict, it does validate the test but it isn't what the docstring says ;)